### PR TITLE
Damage example

### PR DESCRIPTION
**Pull request by TTitscher**
Piece of code to illustrate some uses of the `fenics_helpers` modules.

Have a look at https://github.com/nutofem/fenics_helpers/blob/damage_example/README.md for an impression of the plots.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,3 +4,24 @@
 [![CircleCI](https://circleci.com/gh/nutofem/fenics_helpers/tree/master.svg?style=svg)](https://circleci.com/gh/nutofem/fenics_helpers/tree/master)
 
 Little Helper classes that come up repeatedly when writing solvers in FEniCS
+
+
+# Example: Local damage model
+
+![local damage plot](examples/kappa_plot.png)
+
+The [local damage](examples(local_damage.py) example illustrates the use of `.boundary` to conveniently select the boundaries like
```
Comment (Psirus) [inline README.md]: typo: should probably be `examples/local_damage.py`

File: README.md [hunk 2]
```diff
@@ -4,3 +4,24 @@
...
+
+~~~py
+bot = fh.boundary.plane_at(-10, "y") 
+right = fh.boundary.plane_at(10, "x") 
+~~~
+
+and `.timestepping` is used to reduce the time step size only at critical points in the simulation to accelerate the time integration.
```
Comment (Psirus) [inline README.md]: Maybe just say that it provides *adaptive timestepping*?

File: examples/local_damage.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+
+    return mesh
+
+
+def max(a, b):
+    return 0.5 * (a + b + abs(a - b))
+
+class Mat:
+    E= 20000
```
Comment (Psirus) [inline examples/local_damage.py]: Run black on the file.

File: examples/local_damage.py [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        if d is None:
+            d = self.d
+        return df.sym(df.grad(d))
+    
+    def sigma(self):
+        eps, E, nu = self.eps(), self.mat.E, self.mat.nu
+        lmbda = E * nu / (1. + nu) / (1. - 2. * nu)
+        mu = E / 2. / (1. + nu)
+
+        return 2 * mu * eps + lmbda * df.tr(eps) * df.Identity(2)
```
Comment (Psirus) [inline examples/local_damage.py]: This seems like the only place where the dimension of the `Problem` class is hardcoded. I know this is only an example, but maybe you can do it dimension-independent?

File: examples/local_damage.py [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+            self.plot(disp, load)
+
+    def show(self, fmt="-rx"):
+        self.plot = fh.plotting.AdaptivePlot(fmt)
+
+class Plotter:
+    def __init__(self, model, filename=None):
+        self.model = model
+        if filename is None:
+            filename = "out.xdmf"
```
Comment (Psirus) [inline examples/local_damage.py]: Seems like you could just say `def __init__(self, model, filename="out.xdmf"):`, no?

File: examples/local_damage.py [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+    return solver.solve()
+
+def pp(t):
+    problem.update()
+    ld(t)
+    plot_2d(t)
+
+ts = fh.timestepping.TimeStepper(solve, pp, u=problem.d)
+ts.increase_num_iter = 7
+ts.adaptive(1)
```
Comment (Psirus) [inline examples/local_damage.py]: Maybe say `ts.adaptive(t_end=1.)` to be explicit and avoid the magic number. (With it being an integer, I really didn't know what it was supposed to be.)

**Review by Psirus — APPROVED** [review]

**Comment by TTitscher** [comment]
Do you think that - in our quest of reproducibility - I should automate the image creation?

**Comment by Psirus** [comment]
Not really. This is an example for the functionality of this package, not "scientific output" I would say. Automating that seems to duplicate what you're trying to do in the other project.

**Pull request merged**